src/components/Vendor/UploadOportunity.jsx

```diff
@@ -0,0 +1,559 @@
+      ancho: -1,
+      profundidad: -1,
+      peso: -1,
+
+      filesLoaded: false,
+      isAuthenticated: false
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  getDate(){
+    var today = new Date();
+    var dd = String(today.getDate()).padStart(2, '0');
+    var mm = String(today.getMonth() + 1).padStart(2, '0');
+    var yyyy = today.getFullYear();
+    today = dd + '/' + mm + '/' + yyyy;
+    return today
+  }
+
+  changeDateFormat(inputDate){  // Expects yyyy-mm-dd
+    var newdate = inputDate.split("-").reverse().join("/");
+    return newdate
+  }
+
+  changeDateFormat2(inputDate){  // Expects dd/mm/yyyy
+    var newdate = inputDate.split("/").reverse().join("-");
+    return newdate
+  }
+
+  currentActiveUser() {
+    Auth.currentAuthenticatedUser({})
+      .then(user => {
+        this.setState({
+          vendor: user.attributes,
+          vendor_username: user.username.toLowerCase(),
+          vendor_rol: user.attributes["custom:role"]
+        });
+        console.log("Authenticated User FileUpload");
+        console.log(this.state.vendor)
+      })
+      .catch(err => console.log(err));
+  }
+
+  async componentDidMount() {
+    console.log("LOGUEANDO PARENT PROPS");
+    console.log(this.props);
+    this.currentActiveUser();
+    //console.log(this.state.vendor_username)
+    //this.setState({date_created: this.getDate()});
+  }
+
+  setDimensions(alto, ancho, prof, peso) {
+    this.state.dimensions.height = alto;
+    this.state.dimensions.width = ancho;
+    this.state.dimensions.depth = prof;
+    this.state.dimensions.weight = peso;
+  }
+
+  setAttributes(marca, modelo, color) { 
+    let brand = {};
+    brand.id = 'Marca';                     
+    brand.value = marca;
+    this.state.attributes.push(brand);
+
+    let model = {};
+    model.id = 'Modelo';                     
+    model.value = modelo;
+    this.state.attributes.push(model);
+
+    let colour = {};
+    colour.id = 'Color';                     
+    colour.value = color;
+    this.state.attributes.push(colour)
+}
+
+  handleChange(event) { 
+    const target = event.target;
+    const value = this.assignInputValue(target);
+    const name = target.name;
+
+    this.setState({
+      [name]: value
+    });
+  }
+
+  assignInputValue(target){
+    if (target.type === 'text'){
+      if (target.name == "initial_stock" || target.name == "initial_price"){
+        return target.value
+      }
+      else{
+        return target.value.toUpperCase();
+      }
+    }   
+    if(target.type === 'date'){
+      return this.changeDateFormat(target.value);
+    }
+    else{
+        return target.value
+    } 
+  }
+  
+  getCategories() {
+    const url =
+      'http://proyectoback-tesis.us-west-2.elasticbeanstalk.com/catalog/categories/get_all';
+    fetch(url, {
+      method: 'GET',
+      mode: 'no-cors'
+    })
+    .then(myJson => {
+        console.log('KE LO KE');
```

Comment:
chequear en myjson.status o myjson.code  el codigo de retorno....es algo que dijimos que ibamos a empezar a ver. NICE TO HAVE  no hace falta que lo cambies ahora